spk/owncloud/src/service-setup.sh

```diff
@@ -0,0 +1,507 @@
+
+# ownCloud service setup
+WEB_DIR="/var/services/web_packages"
+# for backwards compatability
+if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 7 ];then
+    WEB_DIR="/var/services/web"
+fi
+if [ -z "${SYNOPKG_PKGTMP}" ]; then
+    SYNOPKG_PKGTMP="${SYNOPKG_PKGDEST_VOL}/@tmp"
+fi
+
+# Others
+OCROOT="${WEB_DIR}/${SYNOPKG_PKGNAME}"
+SQLITE="/bin/sqlite3"
+JQ="/bin/jq"
+SED="/bin/sed"
+SYNOSVC="/usr/syno/sbin/synoservice"
+
+if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+    GROUP="http"
+fi
+
+set_owncloud_permissions ()
+{
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+        DIRAPP=$1
+        DIRDATA=$2
+        echo "Setting the correct ownership and permissions of the files and folders in ${DIRAPP}"
+        # Set the ownership for all files and folders to sc-owncloud:http
+        find -L ${DIRAPP} -type d -print0 | xargs -0 chown ${EFF_USER}:${GROUP} 2>/dev/null
+        find -L ${DIRAPP} -type f -print0 | xargs -0 chown ${EFF_USER}:${GROUP} 2>/dev/null
+        # Use chmod on files and directories with different permissions
+        # For all files use 0640
+        find -L ${DIRAPP} -type f -print0 | xargs -0 chmod 640 2>/dev/null
+        # For all directories use 0750
+        find -L ${DIRAPP} -type d -print0 | xargs -0 chmod 750 2>/dev/null
+        # For external data directory
+        if [ -n "${DIRDATA}" ] && [ -d "${DIRDATA}" ]; then
+            chown -R ${EFF_USER}:${GROUP} ${DIRDATA} 2>/dev/null
+            find -L ${DIRDATA} -type f -print0 | xargs -0 chmod 640 2>/dev/null
+            find -L ${DIRDATA} -type d -print0 | xargs -0 chmod 750 2>/dev/null
+        fi
+        # Set the occ command to executable
+        chmod +x ${DIRAPP}/occ 2>/dev/null
+    else
+        echo "Notice: set_owncloud_permissions() is no longer required on DSM7."
+    fi
+}
+
+exec_occ() {
+    PHP="/usr/local/bin/php74"
+    OCC="${OCROOT}/occ"
+    OCC_ARGS=()
+    for arg in "$@"; do
+        OCC_ARGS+=("$arg")
+    done
+    COMMAND="${PHP} ${OCC} ${OCC_ARGS[@]}"
+    if [ ${SYNOPKG_DSM_VERSION_MAJOR} -lt 7 ]; then
+        OCC_OUTPUT=$(/bin/su "$EFF_USER" -s /bin/sh -c "$COMMAND")
+    else
+        OCC_OUTPUT=$($COMMAND)
+    fi
+    OCC_EXIT_CODE=$?
+    echo "$OCC_OUTPUT"
+    return $OCC_EXIT_CODE
+}
+
+service_prestart ()
+{
+    # Replace generic service startup, fork process in background
+    echo "Starting owncloud-daemon at ${SYNOPKG_PKGDEST}/bin" >> ${LOG_FILE}
+    COMMAND="${SYNOPKG_PKGDEST}/bin/owncloud-daemon"
+    ${COMMAND} >> ${LOG_FILE} 2>&1 &
+    echo "$!" > "${PID_FILE}"
```

Comment:
Why would we need to replace the generic service startup ?
It would seem to me that the command here is pretty simple is it not ? (I may have missed something here)

Reply:
Because the service startup is not actually stating anything (it's just calling a dummy cron config), it needed to have something to have a PID file. As such I took the example from our Wiki - [Custom service process startup](https://github.com/SynoCommunity/spksrc/wiki/Service-Support#custom-service-process-startup) to get the daemon to work. If there is a better way to represent a start/stop on a PHP-based website I'm open to it.

Reply:
If you want I did something of the kind for `tt-rss` (there is an update daemon script which `really` needs being started to have RSS feeds updated)

Reply:
@smaarn, I tried using the `SERVICE_COMMAND` however the package failed to start (even when I removed the while loop in the daemon). In comparing with `tt-rss`, the daemon file there included the `--daemon` option as part of the execution. Unfortunately, the OCC utility does not have such an option. I've since reverted to the original code so that the package start/stop can function.